tests/strategies/test_collaborative.py

```diff
@@ -0,0 +1,370 @@
+import multiprocessing as mp
+import os
+import time
+from typing import Any
+from unittest import mock
+from unittest.mock import PropertyMock
+
+import pytest
+import requests
+import torch
+from torch.optim import Optimizer
+
+import pytorch_lightning as pl
+from pytorch_lightning.plugins.environments.lightning_environment import find_free_network_port
+from pytorch_lightning.strategies import CollaborativeStrategy
+from pytorch_lightning.strategies.collaborative import HiveMindScheduler
+from pytorch_lightning.utilities import _HIVEMIND_AVAILABLE
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+from pytorch_lightning.utilities.types import STEP_OUTPUT
+from tests.helpers import BoringModel
+from tests.helpers.runif import RunIf
+
+if _HIVEMIND_AVAILABLE:
+    import hivemind
+
+
+@mock.patch("pytorch_lightning.strategies.collaborative._HIVEMIND_AVAILABLE", False)
+def test_raise_exception_if_hivemind_unavailable():
+    """Test that we raise an exception when Hivemind is not available."""
+    with pytest.raises(MisconfigurationException, match="you must have Hivemind installed"):
+        CollaborativeStrategy(target_batch_size=1)
+
+
+@RunIf(hivemind=True)
+@mock.patch("hivemind.DHT", autospec=True)
+def test_strategy(mock_dht):
+    strategy = CollaborativeStrategy(target_batch_size=1)
+    trainer = pl.Trainer(strategy=strategy)
+    assert trainer.strategy == strategy
+
+
+@RunIf(hivemind=True)
+@mock.patch("hivemind.DHT", autospec=True)
+@mock.patch("pytorch_lightning.strategies.collaborative.DHTManager._get_peers", autospec=True)
+@pytest.mark.parametrize(
+    "initial_peers,peer_endpoint",
+    [(["TEST"], None), (None, "localhost:153")],
+)
+def test_logging_disabled_when_second_peer(mock_dht, mock_http, initial_peers, peer_endpoint):
+    """Test when we are a second peer (passing initial peers or peer endpoint) we warn the user that
+    logging/checkpointing will be disabled."""
+    with pytest.warns(UserWarning, match="This machine is not a persistent machine"):
+        CollaborativeStrategy(target_batch_size=1, initial_peers=initial_peers, peer_endpoint=peer_endpoint)
+
+
+@RunIf(hivemind=True)
+@mock.patch.dict(
+    os.environ,
+    {"HIVEMIND_MEMORY_SHARING_STRATEGY": "file_descriptor", "PORT": str(find_free_network_port())},
```

Comment:
noob question: Should this env var be PL-specific one which gets read by `DHTManager` class?
```suggestion
    {"HIVEMIND_MEMORY_SHARING_STRATEGY": "file_descriptor", "PL_PORT": str(find_free_network_port())},
```

Reply:
Yes it should! Great catch :)